### DISCUSSION TITLE
Farnsworth Ability

### DISCUSSION THREAD

**Discussion by scotthibbs** [w7sst/MorseRunner#114] [Ideas]
At our Eatn' before the Meetn' outing last night, Walt WB4ZUT, asked if we had considered a Farnsworth setting as he understands CW at 10wpm easy but would like to speed up to 20wpm. 

We both discussed that this program isn't necessarily a CW learning program (I suggest Morse Machine at http://www.g4ilo.com/morse-machine.html). I also mentioned that I wasn't privy to how the program constructs the responses. I know it uses a list but not how it takes the list of calls and constructs the CW. 

So provided those two facts are true: learning program for pile ups but not learning CW, and that it might be impossible; would this be something to consider after we get all the contests world wide added?

**Comment by w7sst** [comment]
I can see adding some Farnsworth spacing when generating the CW sounds. This would be easier to do than adding CW weights as suggested in another reading. Yes, this is something to consider. Adding Farnsworth support would require us to have a working Settings dialog so these other advanced settings could be supported. I also agree that adding this after we get the contests we wanted added up and running. We also want to integrate any good-ideas from the other MR projects (work from prior authors).

I also agree that the original goal of MR is to operate as a Contest Simulator. Some of these other options add to the usability, but we want to be sure that we are focusing on the contest simulator aspect. this includes generating good-sounding CW at faster rates.

**Comment by tekenny** [comment]
personally I think we should focus on MR’s mission of contest prep, Farnsworth doesn’t exist in tests and there are many apps, web sites and cw training elsewhere for improving copying cw… I am a member of LICW which is great. I’ve also heard many complain that they can copy 32/20 Farnsworth but when they get on the air can’t do 20/20 so they created a new hurdle for themselves. Don’t get me wrong I sometimes use Farnsworth in my cw studies. So I do think for now we focus on bugs, enhancements, contests and tech debt before Farnsworth, my two cents :-)

**Comment by tekenny** [comment]
PS - we may want to add a list of CW learning resources: software, web sites, training, clubs, etc to the readme file?

**Reply by K6OK** [reply]
I concur.  Suggest add a statement such as, "Morse Runner Community Edition is not intended as a general CW learning tool.  Its primary function is a CW contest simulator.  Some CW learning resources include <insert LCWO, G4FON, ARRL bulletins, CW Academy, etc>"

**Reply by w7sst** [reply]
> Suggest add a statement such as, "Morse Runner Community Edition is not intended as a general CW learning tool. Its primary function is a CW contest simulator. Some CW learning resources include <insert LCWO, G4FON, ARRL bulletins, CW Academy, etc>"

I agree with adding the above statement. We should focus on contest simulation and adding additional contests. Our sound quality goal should be no worse than 1.68. I think 1.68 should be our quality standard.

I'd like to see it be an easy effort to add an additional contest. This allows the community to help add more contests. Integration with popular contest logging software programs (e.g. N1MM+) is also important.

Perhaps long term, we can look into other options to help as a practice tool. 

I would also add "RufZp" to @Yagi-Berra above list.

Thanks to all for such good input on direction of the MRCE project. This has been good input on helping us refine our goals and direction.

**Comment by w7sst** [comment]
I agree. Let's focus on MR's original purpose as a contest simulator.

**Comment by w7sst** [comment]
Update on Farnsworth support... After several weeks of work, I have a working prototype of Farnsworth support for the SST Contest. I think this will work fine for the SST Contest only. I will be submitting a code review for this work soon and then release a 1.83 pre-release build soon so everyone can give this a test drive.

Update... As I re-read this discussion, I see that I may have taken MR slightly away from the recommendation of this discussion. However, there is significant interest from the CWOPS Community to use MRCE as part of their training curriculum w.r.t. the SST Contest. They felt the Farnsworth support would be a good addition and the compromise we are implementing in MRCE is to support Farnsworth only in the SST Contest.

Issue #188 has been created for this implementation.

**Comment by w7sst** [comment]
I have completed my work on Farnsworth support for the K1USN SST Contest. Please see #188 for details. It is in code review [here](https://github.com/w7sst/MorseRunner/pull/196). This is the last major change for the upcoming v1.83 version. Time to get this March release wrapped up.

@scotthibbs
